src/features/options/useGetOptions.ts

```diff
@@ -70,33 +76,87 @@ const compareOptionAlphabetically =
     return sortOrder === 'asc' ? comparison : -comparison;
   };
 
-export function useGetOptions<T extends ValueType>(props: Props<T>): OptionsResult {
-  const { node, options, optionsId, secure, removeDuplicates, source, mapping, queryParameters, sortOrder, metadata } =
-    props;
+export function useGetOptions<T extends ValueType>(props: Props<T>): OptionsResult<T> {
+  const {
+    node,
+    options,
+    optionsId,
+    secure,
+    removeDuplicates,
+    source,
+    mapping,
+    queryParameters,
+    sortOrder,
+    dataModelBindings,
+    valueType,
+  } = props;
+  const value = FD.usePickFreshString(dataModelBindings?.simpleBinding);
+  const setDataUpstream = FD.useSetForBindings(dataModelBindings);
   const sourceOptions = useSourceOptions({ source, node });
   const staticOptions = optionsId ? undefined : options;
   const { data: fetchedOptions, isFetching } = useGetOptionsQuery(optionsId, mapping, queryParameters, secure);
   const calculatedOptions = sourceOptions || fetchedOptions?.data || staticOptions;
   const { langAsString } = useLanguage();
   const selectedLanguage = useCurrentLanguage();
 
-  const setMetadata = metadata?.setValue;
   const downstreamParameters: string = fetchedOptions?.headers['altinn-downstreamparameters'];
   useEffect(() => {
-    if (!!setMetadata && downstreamParameters) {
-      setMetadata(downstreamParameters);
+    if (dataModelBindings && 'metadata' in dataModelBindings && dataModelBindings.metadata && downstreamParameters) {
```

Comment:
Do you have to check all of these specifically or would this be covered by `dataModelBindings?.metadata && downstreamParameters`

Reply:
Yup, sadly. Since the type is `IDataModelBindingsOptionsSimple | IDataModelBindingsSimple` and the `metadata` key does not exist at all in `IDataModelBindingsSimple`. I blame TypeScript for that one.